### PR TITLE
CVE-2011-1198

### PR DESCRIPTION
**Pull request by aaa1008**
Completed research and verified existing information.

### CODE REVIEW

File: cves/CVE-2011-1198.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: Out-of-bounds Read
```
Comment (andymeneely) [inline cves/CVE-2011-1198.yml]: Need the number for this.

**Comment by ath9902** [comment]
CWE number should be 125 which is out of bounds read. You could add a note to defense in depth for lessons learned that developers need to do careful resource checking and management.

File: cves/CVE-2011-1198.yml [hunk 2]
```diff
@@ -100,20 +104,25 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    I looked in the Chromium bug reports, read the dicussions, the git logs,
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1198.yml]: Remove first-person references and just say something like, "No information on how bug was discovered".

File: cves/CVE-2011-1198.yml [hunk 3]
```diff
@@ -125,10 +134,10 @@ interesting_commits:
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
   answer:
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1198.yml]: Demonstrate that you completed this section by explaining what happened between the VCCs and the fix.

File: cves/CVE-2011-1198.yml [hunk 4]
```diff
@@ -159,38 +168,38 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1198.yml]: Please add a note describing why Defense in Depth applies for this specific vulnerability.

File: cves/CVE-2011-1198.yml [hunk 5]
```diff
@@ -159,38 +168,38 @@ lessons:
...
   native_wrappers:
-    applies: 
-    note: 
+    applies:
+    note:
   distrust_input:
-    applies: 
-    note: 
+    applies: true
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1198.yml]: Same note as above

File: cves/CVE-2011-1198.yml [hunk 6]
```diff
@@ -203,4 +212,7 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer:
+    Unaccounting for threads in addition to out-of-bound errors depending on
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1198.yml]: Add more content and detail to your analysis.  Was this a coding mistake, a design mistake, or something else?  Does this fix look sufficient?  Was there anything else the devs could have done to mitigate this issue or prevent it from occurring in the first place? etc.

File: cves/CVE-2011-1198.yml [hunk 7]
```diff
@@ -203,4 +212,7 @@ mistakes:
...
+    thread count. Additionally, this was a high security bug that required the
+    team to fix it early on rather than waiting for another release.
```
Comment (AdamVD) [inline cves/CVE-2011-1198.yml]: This is an interesting point. From a brief examination of the issue page, it seems that they wanted to fix it quickly for the M9 release, but nobody started working on it until later and it made it into M10. Maybe there weren't enough developers to cover everything that should have been done for the M9 release? This seems to be something you can include in the 'major_events' section.

**Comment by AdamVD** [comment]
I'll give this +2 based on it being a high-severity issue it didn't make it into the next release after being found.

File: cves/CVE-2011-1198.yml (hunk 1 above)
Comment (aaa1008) [reply cves/CVE-2011-1198.yml]: number added

File: cves/CVE-2011-1198.yml [hunk 8]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
...
   If you are manually editing this file, then you are "curating" it. Set the
   entry below to "true" as soon as you start. This will enable additional
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
 announced_instructions: |
   Was there a date that this vulnerability was announced to the world? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE date. A good
```
[the remark this answers is not in the document]
Comment (aaa1008) [reply cves/CVE-2011-1198.yml]: timestamp removed. Thanks.

File: cves/CVE-2011-1198.yml (hunk 7 above)
Comment (aaa1008) [reply cves/CVE-2011-1198.yml]: Thanks for the comment. That's a possible explanation for why it was left unresolved considering the vulnerabilities level of severity. However, I'm unsure if there'a way to find out whether there were enough developers to cover all security risks at that time or not.

**Pull request merged**